dev-tools/xtask/src/main.rs

```diff
@@ -213,6 +237,86 @@ fn cmd_check_workspace_deps() -> Result<()> {
     Ok(())
 }
 
+fn cmd_verify_library() -> Result<()> {
+    let metadata = load_workspace()?;
+    let mut config_path = metadata.workspace_root;
+    config_path.push("library-verification.toml");
+    let config = read_library_verification_toml(&config_path)?;
+
+    let mut offenders = Vec::new();
+    for entry in fs::read_dir(&metadata.target_directory)
+        .with_context(|| format!("read_dir {:?}", &metadata.target_directory))?
+    {
+        let entry = entry.context("dirent")?;
+        if !entry.file_type()?.is_dir() {
+            continue;
+        }
+
+        // Only search for binaries in target/release and target/debug
+        if entry.file_name() != "release" && entry.file_name() != "debug" {
+            continue;
+        }
+
+        for binary in metadata
+            .packages
+            .iter()
+            .flat_map(|p| &p.targets)
+            .filter(|t| t.kind == vec!["bin"])
+            .map(|x| &x.name)
+        {
+            let mut path = entry.path();
+            path.push(binary);
+
+            if !path.is_file() {
+                continue;
+            }
+
+            let command = Command::new("elfedit")
+                .arg("-o")
+                .arg("simple")
+                .arg("-r")
+                .arg("-e")
+                .arg("dyn:tag NEEDED")
+                .arg(&path)
+                .output()
+                .context("exec elfedit")?;
+
+            if !command.status.success() {
+                bail!(
+                    "Failed to execute elfedit successfully {}",
+                    command.status
+                );
+            }
+
+            let stdout = String::from_utf8(command.stdout)?;
+            // `elfedit -o simple -r -e "dyn:tag NEEDED" /file/path` will return
+            // a new line seperated list of required libraries so we walk over
+            // them looking for a match in our configuration file. If we find
+            // the library we make sure the binary is allowed to pull it in via
+            // the whitelist.
+            for library in stdout.lines() {
+                if let Some(whitelist) = config.libraries.get(library.trim()) {
+                    if !whitelist.binaries.contains(binary) {
+                        offenders.push(anyhow::anyhow!(
+                            "{binary} NEEDS {library} but \
+                            it is not whitelisted"
+                        ));
+                    }
+                }
+            }
+        }
+    }
+
+    if !offenders.is_empty() {
+        bail!(
+            "Found the following issues: {offenders:#?}\n\nIf you believe any \
+            of these issues are wrong please update the whitelist."
```

Comment:
Instead of using the `Debug` impl, it would be best to generate a nicely formatted list as a string and print it out here.